src/jdk.incubator.vector/share/classes/jdk/incubator/vector/ByteVector.java

```diff
@@ -1904,6 +1904,30 @@ M compareTemplate(Class<M> maskType, Comparison op, Vector<Byte> v) {
             });
     }
 
+    /*package-private*/
+    @ForceInline
+    final
+    <M extends VectorMask<Byte>>
+    M compareTemplate(Class<M> maskType, Comparison op, Vector<Byte> v, M m) {
+        Objects.requireNonNull(v);
+        Objects.requireNonNull(m);
+        ByteSpecies vsp = vspecies();
+        ByteVector that = (ByteVector) v;
+        that.check(this);
```

Comment:
```suggestion
        ByteVector that = (ByteVector) v;
        that.check(this);
        m.check(maskClass, this);
```

And similarly for the non-mask accepting method (we don't need the explicit null check and vsp is not used).